### PR TITLE
add DataFormatter

### LINKED ISSUE #810: [FEAT][CV] Add Formatter for data
**Is your feature request related to a problem? Please describe.**
Currently the package supports formatters for labels and predictions to help ensure they come in a fixed format that the checks can accept. Such formatters are missing for the data itself - the images. 

**Describe the solution you'd like**
Add formatter for data.

### PR DESCRIPTION
**Pull request by noamzbr**
#### Reference Issues/PRs

Closes #810 

#### What does this implement/fix? Explain your changes.
Added data formatter as member of the VisionDataset to convert data to a displayable format. 

#### Any other comments?
The selected format was an iterable of 3D numpy arrays, with [H, W, C] and values between 0 and 255. Please comment if you think we should use a different format.

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/vision/utils/data_formatters.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        Returns
+        -------
+        Optional[str]
+            None if the data is valid, otherwise a string containing the error message.
+
+        """
+
+        batch_data = self(batch_data)
+        print(batch_data.shape)
```
Comment (ItayGabbay) [inline deepchecks/vision/utils/data_formatters.py:44]:
```suggestion

```

File: deepchecks/vision/utils/data_formatters.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+class DataFormatter:
+    """
+    Class for formatting the image data outputted from the dataloader to the required format for check displays.
+
+    Parameters
+    ----------
+    data_formatter : Callable
+        Function that takes in a batch of data and returns the data in the following format (a batch of cv2 images):
+        A 4D numpy array, with the first dimension being the batch size, the second being the image height (y axis), the
```
Comment (ItayGabbay) [inline deepchecks/vision/utils/data_formatters.py:15]: Maybe write explicitly [M, H, W, C]

File: deepchecks/vision/utils/data_formatters.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        -------
+        Optional[str]
+            None if the data is valid, otherwise a string containing the error message.
+
+        """
+
+        batch_data = self(batch_data)
+        print(batch_data.shape)
+
+        if not isinstance(batch_data, np.ndarray):
```
Comment (ItayGabbay) [inline deepchecks/vision/utils/data_formatters.py:46]: What about torch.Tensor?

**Review by ItayGabbay — COMMENTED** [review]
Looks good. added a few cmnts

File: deepchecks/vision/utils/data_formatters.py (hunk 3 above)
Comment (noamzbr) [reply deepchecks/vision/utils/data_formatters.py:46]: Tensors are mostly common at the model input form, which is mostly [C, H, W] and values around 0 with std 1. Think that if the user needs to covert back from that to a displayable format, it's also reasonable to convert back to numpy.

**Review by matanper — APPROVED** [review]

**Pull request merged**